### DISCUSSION TITLE
Prevent functions from being made `Public` when it isn't used in another file

### DISCUSSION THREAD

**Discussion by France30** [eslint/eslint#14847] [Q&A]
I'm trying to figure out how to prevent functions from being made `Public` when it isn't used in another file.

I tried looking for an eslint rule for this, but I couldn't seem to find one and the closest was using `explicit-member-accessibility`

So far I've tried to work-around it by doing this 

```
"@typescript-eslint/explicit-member-accessibility": [
	"warn",
	{
		"accessibility": "explicit",
		"overrides": {
			"accessors": "off",
			"constructors": "no-public",
			"methods": "explicit",
			"properties": "off",
			"parameterProperties": "off"
		}
	}
]
```
            
But the problem with it is that it forces you to write `Public` before every method.
This would also require editing a lot of code to an already existing codebase.

Is there a way to solve this problem or an eslint rule that I might have missed?

**Comment by nzakas** [comment]
There is nothing in the core for this. If you’re using typescript-eslint, then I’d suggest asking them for help. (We don’t maintain that project)

**Reply by France30** [reply]
Alright, thank you
